src/java/org/apache/cassandra/db/virtual/GossipInfoTable.java

```diff
@@ -76,10 +87,10 @@ final class GossipInfoTable extends AbstractVirtualTable
     public DataSet data()
     {
         SimpleDataSet result = new SimpleDataSet(metadata());
-        for (Map.Entry<InetAddressAndPort, EndpointState> entry : Gossiper.instance.endpointStateMap.entrySet())
+        for (Map.Entry<InetAddressAndPort, EndpointState> entry : endpointStateMapSupplier.get().entrySet())
         {
             InetAddressAndPort endpoint = entry.getKey();
-            EndpointState localState = entry.getValue();
+            EndpointState localState = new EndpointState(entry.getValue());
```

Comment:
Maybe we can add a brief comment here saying that we are snapshotting the `EndpointState`s obtained from Gossiper because they can be concurrently modified by `LoadBroadcaster`.

Reply:
ok i ll add it.